### ISSUE TITLE
Abilitz to unset the `payment_method_types` parameter

### ISSUE THREAD

**Issue by jacob418** [laravel/cashier-stripe#1379]
Stripe offers the ability to [manage supported payment methods from the dashboard](https://stripe.com/docs/payments/dashboard-payment-methods#update). However within `\Laravel\Cashier\Checkout::create` this parameter is hardcoded. There should be an option, to make Cashier obmit this parameter completely so the full potential of stripe can be used. For example: `\Laravel\Cashier\Cashier::configurePaymentMethodsViaDashboard(true)`.

**Comment by driesvints** [comment]
Yes, we should make this configurable somehow. Maybe adding an `array_filter` around the options so you can pass in `'payment_method_types' => []` might be a good intermediate fix.

**Comment by ffeytons** [comment]
For a project of mine, I have the exact same need coming.
@driesvints You have set the "help wanted" flag, I'm willing to contribute for this feature -- is there some place in which you synchronize / assign tasks for this project?

**Comment by driesvints** [comment]
Would appreciate a PR to 13.x for the array filter thing 👍

**Comment by ffeytons** [comment]
Alright, will work on it somewhere this week!

**Comment by ffeytons** [comment]
Hi,

I've just done a few tests, passing `'payment_method_types' => []` is already working, so there is no intermediate fix needed over here.

This is because `Checkout::create` in Cashier's code does this:

```php
$session = $owner->stripe()->checkout->sessions->create(array_merge([
            'customer' => $customer->id,
            'mode' => 'payment',
            'success_url' => $sessionOptions['success_url'] ?? route('home').'?checkout=success',
            'cancel_url' => $sessionOptions['cancel_url'] ?? route('home').'?checkout=cancelled',
            'payment_method_types' => ['card'],
        ], $sessionOptions));
```

The parameters we pass to Cashier from the app are stored in `$sessionOptions`.
As per `array_merge` documentation, if both arrays have the same key, the value of the later overwrites the former.

@jacob418 could you confirm passing `'payment_method_types' => []` to the `checkout([...])` method works for you?

**Comment by driesvints** [comment]
@ffeytons hah, you're right, that should indeed already work.

**Comment by jacob418** [comment]
> The parameters we pass to Cashier from the app are stored in `$sessionOptions`. As per `array_merge` documentation, if both arrays have the same key, the value of the later overwrites the former.
> 
> @jacob418 could you confirm passing `'payment_method_types' => []` to the `checkout([...])` method works for you?

I can confirm that the proposed empty array parameter (`'payment_method_types' => [],`) will be accepted by Stripe API and seems to enable the dashboard-managed payment methods. I should have tried it in the first place but only came up with `null` wich was rejected by stripe API. 🤦‍♂️ Ty for your help.
